digitalidentity/service.go

```diff
@@ -33,14 +36,47 @@ func CreateShareSession(httpClient requests.HttpClient, shareSession *ShareSessi
 		HTTPMethod: http.MethodPost,
 		BaseURL:    apiUrl,
 		Endpoint:   endpoint,
-		Headers:    nil,
+		Headers:    requests.AuthHeader(clientSdkId),
 		Body:       payload,
 	}.Request()
 	if err != nil {
 		return share, err
 	}
 
-	response, err := requests.Execute(httpClient, request, ShareURLHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
+	response, err := requests.Execute(httpClient, request, ShareSessionHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return share, err
+	}
+
+	defer response.Body.Close()
+
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return share, err
+	}
+
+	err = json.Unmarshal(responseBytes, &share)
+
+	return share, err
+}
+
+// GetSession get session info using the supplied sessionID
+func GetSession(httpClient requests.HttpClient, sessionID string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ShareSession, err error) {
+	endpoint := identitySesssionCreationEndpoint
+	headers := requests.AuthHeader(clientSdkId)
+	request, err := requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    apiUrl,
+		Endpoint:   endpoint,
+		Headers:    headers,
+	}.Request()
+	if err != nil {
+		return share, err
+	}
+
+	response, err := requests.Execute(httpClient, request, ShareSessionHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
```

Comment:
i think we need to update this `requests` package to support the error codes connect send back, some of these are really useful and we should differentiate between them and give the RP an exported error var they can switch on and handle. 

e.g. from looking at [this](https://lampkicking.atlassian.net/wiki/spaces/CON/pages/3756458012/Share+v2+Flows) `404` `UNKNOWN_SESSION` is invalid session id which would be useful for a RP to know when integrating with sdk, as opposed to the `ShareSessionHTTPErrorMessages` where `404` is mapped to `Application was not found`?

Reply:
@mehmet-yoti are you still working on this?

Reply:
Yes I am doing some changes on that.

Reply:
i don't see the changes for this from the newest commits @mehmet-yoti - feel free to ping me on slack if you wanna chat about it abit quicker